src/BackupExplorer/Parser/BackupParserImpl.cs

```diff
@@ -51,6 +51,39 @@ public BackupParserImpl(string backupChainPath, string codePackagePath)
             this.transactionChangeManager = new TransactionChangeManager( this.reliabilitySimulator);
         }
 
+        public BackupParserImpl(string backupChainPath, string codePackagePath, ILog log, string workFolderPath)
+        {
+            this.backupChainPath = backupChainPath;
+            this.codePackage = new CodePackageInfo(codePackagePath);
+            
+            if (String.IsNullOrEmpty(workFolderPath))
+            {
+                this.workFolder = Path.Combine(Directory.GetCurrentDirectory(), Guid.NewGuid().ToString()); 
+            }
+            else
+            {
+                this.workFolder = Path.Combine(workFolderPath, Guid.NewGuid().ToString());
+            }
+            
+            if (log != null) BackupParserImpl.log = log;
+
+            Console.WriteLine("Work Folder : {0}", this.workFolder);
+
+            Directory.CreateDirectory(this.workFolder);
+            this.reliabilitySimulator = new ReliabilitySimulator(
+                this.workFolder,
+                new Uri("fabric:/rcbackupapp/rcbackupservice"),
+                DateTime.UtcNow.ToFileTimeUtc(),
+                DateTime.UtcNow.ToFileTimeUtc(),
+                this.OnDataLossCallback,
+                this.CreateStateProvider);
+            this.reliabilitySimulator.CreateReplica(true, false);
+
+            this.stateManager = new StateManager(reliabilitySimulator);
+            this.seenFirstTransaction = false;
+            this.transactionChangeManager = new TransactionChangeManager(this.reliabilitySimulator);
```

Comment:
Instead of duplicating all this can we just call the other method which does it already ?

Reply:
Updated to make it more modular